### PR TITLE
GridModel to accept a `defaultCol`

### PR DESCRIPTION
**Pull request by YanaNirvana**
Half address Issue #1216: defaultCol config to be applied to all columns, except where otherwise specified
(co dev with jhaynes)

### CODE REVIEW

**Comment by amcclain** [comment]
Thanks for getting this in - appreciate it. 

On reflection, I am not sure about the name `defaultCol` - the sense of that doesn't seem quite right.  How about `colDefaults`?

File: cmp/grid/GridModel.js [hunk 1]
```diff
@@ -191,6 +195,7 @@ export class GridModel {
 
         Object.assign(this, rest);
 
+        this.defaultCol = withDefault(defaultCol, {});
```
Comment (lbwexler) [inline cmp/grid/GridModel.js]: In this case, could do the defaulting in the constructor for self-documenting....also more in-line with the other two defaults.

File: cmp/grid/GridModel.js [hunk 2]
```diff
@@ -115,6 +116,8 @@ export class GridModel {
     /**
      * @param {Object} c - GridModel configuration.
      * @param {Object[]} c.columns - {@link Column} or {@link ColumnGroup} configs
+     * @param {Object} [c.defaultCol] = Column configs to be set on all columns.  If a Column has
```
Comment (lbwexler) [inline cmp/grid/GridModel.js]: I feel that this comment states the obvious regarding what "wins out".  That's fine, but  I think the less obvious point that we should clarify is that the merging is done "deeply"...  I would not assume that as an API user and would be glad to have it clarified....

**Comment by lbwexler** [comment]
Also agree with Anselm's proposed name change.  This will be a nice pickup!

**Pull request merged**